fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryLockFactory.java

```diff
@@ -210,6 +212,13 @@ private void fileLockCheckNewLock(byte[] val, long nowMillis) {
         }
 
         private void fileLockClearFlushAndClose(boolean isRecovery) {
+            synchronized (clearingLockNowLock) {
```

Comment:
Teamscale is reporting that this test is uncovered: https://fdb.teamscale.io/activity/merge-requests/foundationdb-fdb-record-layer/FoundationDB%2Ffdb-record-layer%2F3714
Given that teamscale is not reporting back to PRs right now, it's possible that it is a matter of closing.

This could be an indication that some lower-level tests of `FDBDirectoryLockFactory` are needed, particularly covering the parallelism indicated in the PR description.